R/download.R

```diff
@@ -82,72 +69,99 @@ download_hydat <- function(dl_hydat_here = NULL, ask = TRUE) {
   httr::stop_for_status(x)
   new_hydat <- substr(gsub("^.*\\Hydat_sqlite3_", "",
                            httr::content(x, "text")), 1, 8)
+  #Make the download URL
+  url <- paste0(base_url, "Hydat_sqlite3_", new_hydat, ".zip")
+  response = httr::HEAD(url)
+  size <- round(as.numeric(httr::headers(response)[["Content-Length"]])/1000000, 0)
 
+  
   ## Do we need to download a new version?
-  if (new_hydat == existing_hydat) {
-    handle_error(stop(not_done(paste0("The existing local version of hydat, published on ",
-                lubridate::ymd(existing_hydat),
-                ", is the most recent version available."))))
+  if (new_hydat == existing_hydat & ask) { #DB exists and no new version
+    dl_overwrite <- ask(paste0("The existing local version of HYDAT, published on ", lubridate::ymd(existing_hydat), ", is the most recent version available.  \nDo you wish to overwrite it?  \nDownloading HYDAT could take up to 10 minutes (", size, " MB)."))
   } else {
-    info(paste0("Downloading version of HYDAT created on ", crayon::blue(lubridate::ymd(new_hydat))))
+    dl_overwrite <- TRUE
   }
 
-  url <- paste0(base_url, "Hydat_sqlite3_", new_hydat, ".zip")
-  
-
-  ## temporary path to save
-  tmp <- tempfile("hydat_", fileext = ".zip")
-
-  ## Download the zip file
-  res <- httr::GET(url, httr::write_disk(tmp), httr::progress("down"), 
-                   httr::user_agent("https://github.com/ropensci/tidyhydat"))
-  on.exit(file.remove(tmp), add = TRUE)
-  httr::stop_for_status(res)
-  
-  if(file.exists(tmp)) info("Extracting HYDAT")
-  
-  
-  utils::unzip(tmp, exdir = dl_hydat_here, overwrite = TRUE)
-  
-  ## rename to consistent name
-  file.rename(
-    list.files(dl_hydat_here, pattern = "\\.sqlite3$", full.names = TRUE),
-    hydat_path
-  )
-  
-  
-  if (file.exists(hydat_path)){
-    congrats("HYDAT successfully downloaded")
-  } else(not_done("HYDAT not successfully downloaded"))
-  
-  hy_check()
-  
-  invisible(TRUE)
-}
-
-hy_check <- function(hydat_path = NULL) {
-  con <- hy_src(hydat_path)
-  on.exit(hy_src_disconnect(con), add = TRUE)
-  
-  have_tbls <- dplyr::src_tbls(con)
+  if (!dl_overwrite){
+    info("HYDAT is updated on a quarterly basis, check again soon for an updated version.")
+  }
   
-  tbl_diff <- setdiff(hy_expected_tbls(), have_tbls)
-  if (!rlang::is_empty(tbl_diff)) {
-    red_message("The following tables are missing from HYDAT")
-    red_message(paste0(tbl_diff, "\n"))
+  if (new_hydat != existing_hydat & ask) { #New DB available or no local DB at all
```

Comment:
Can you move this comment to above this?

Reply:
Not too sure what's going on here. Line numbers do not match, and the suggested add of line 89 above is already there. Also, are you sure you want to delete the statement if (!rlang::is_empty(tbl_diff...? If so there are a few other related lines to get rid of.

Reply:
Hmm... I just meant can you move 
> #New DB available or no local DB at all

on to a new line